### PR TITLE
feat(oracle): use TABLE(:bind_array) for bulk get chunks >= 900 keys

### LINKED ISSUE #4041: Oracle state store bulk querying improvements
### Problem
Currently the Oracle state store component will perform a single, unbounded retrieval of multiple keys using a(n) SQL `IN` condition.

For small numbers of keys, this is probably fine.  When dealing with larger numbers of items however, this approach can present some challenges:

- Can be rough on indexes
- Difficult to optimize
- (Might be difficult for execution plans?)

### Suggested improvements

While our general advice might be to suggest that users perform chunking on their end, there may still be a case for configurable chunking dapr-side as well:

- Allow a maximum page/chunk size to be configured
- When total number of keys exceeds the configured page size, dapr internally chunks
  - So, if page size is 100 and you have 1000 keys, it will do 100 keys 10 times...
- Perform chunking in series, not in parallel (this is important!)
- While we're here, it would be a good idea to do this Oracle specific tuneup to the query syntax: `WHERE id IN (SELECT COLUMN_VALUE FROM TABLE(:bind_array))`

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by D2758695161**
## Summary
Implements the optimization described in dapr/components-contrib#4041.

### What changed
In ulkGetChunk, when the number of keys is >= 900, the query now uses Oracle's TABLE(:bind_array) syntax instead of the standard IN (:1, :2, ...) syntax:

\\\sql
SELECT key, value, binary_yn, etag, expiration_time
FROM state_table
WHERE key IN (SELECT COLUMN_VALUE FROM TABLE(:1))
\\\

This passes goora.StringArray(keys) as a bind variable, leveraging the go_ora driver's native Oracle ARRAY support.

### Threshold
- **< 900 keys**: standard IN-list with individual bind variables (existing behavior)
- **>= 900 keys**: TABLE(:bind_array) with StringArray bind (new optimization)

The 900-key threshold leaves a safe margin below Oracle's 1000-expression hard limit.

### Testing
Existing unit tests cover chunking behavior and edge cases. The existing mock-based tests should continue to pass as the logic path is determined at runtime based on key count.

Closes #4041

### CODE REVIEW

File: state/oracledatabase/oracledatabaseaccess.go [hunk 1]
```diff
@@ -408,22 +408,36 @@ func (o *oracleDatabaseAccess) BulkGet(ctx context.Context, req []state.GetReque
...
+
+	var rows *sql.Rows
+	var err error
 
-	rows, err := o.db.QueryContext(ctx, query, params...)
+	if useTableFunc {
+		// Oracle's TABLE(:bind_array) syntax binds a string array as an Oracle
+		// collection type, avoiding the 1000-expression IN-list limit.
+		query := "SELECT key, value, binary_yn, etag, expiration_time FROM " + o.metadata.TableName + " WHERE key IN (SELECT COLUMN_VALUE FROM TABLE(:1)) AND (expiration_time IS NULL OR expiration_time > systimestamp)"
+		rows, err = o.db.QueryContext(ctx, query, goora.StringArray(keys))
```
Comment (javier-aliaga) [inline state/oracledatabase/oracledatabaseaccess.go:427]: goora.StringArray(keys) does this exist?
